SQL/New_patches/2022-05-17-add-scan_type_identification-table-refactor-mri_protocol-part-1.sql

```diff
@@ -0,0 +1,105 @@
+    DROP CONSTRAINT `FK_mri_protocol_group_target_1`;
+ALTER TABLE scan_type_parameter_group_target
+    ADD CONSTRAINT `FK_scan_type_parameter_group_target_1`
+        FOREIGN KEY (`ScanTypeParameterGroupID`)
+            REFERENCES `scan_type_parameter_group` (`ScanTypeParameterGroupID`);
+
+ALTER TABLE scan_type_parameter_group_target
+    DROP CONSTRAINT `FK_mri_protocol_group_target_2`;
+ALTER TABLE scan_type_parameter_group_target
+    ADD CONSTRAINT `FK_scan_type_parameter_group_target_2`
+        FOREIGN KEY (`ProjectID`) REFERENCES `Project` (`ProjectID`);
+
+ALTER TABLE scan_type_parameter_group_target
+    DROP CONSTRAINT `FK_mri_protocol_group_target_3`;
+ALTER TABLE scan_type_parameter_group_target
+    ADD CONSTRAINT `FK_scan_type_parameter_group_target_3`
+        FOREIGN KEY (`SubprojectID`) REFERENCES `subproject` (`SubprojectID`);
+
+
+
+-- Add ScannerID and CenterID to scan_type_identification_group_target
+
+ALTER TABLE scan_type_parameter_group_target
+    ADD COLUMN ScannerID INT(10) UNSIGNED DEFAULT NULL,
+    ADD KEY `FK_scan_type_parameter_group_target_4` (`ScannerID`),
+    ADD CONSTRAINT `FK_scan_type_parameter_group_target_4`
+        FOREIGN KEY (`ScannerID`) REFERENCES `mri_scanner` (`ID`);
+
+ALTER TABLE scan_type_parameter_group_target
+    ADD COLUMN CenterID INT(10) UNSIGNED DEFAULT NULL,
+    ADD KEY `FK_scan_type_parameter_group_target_5` (`CenterID`),
+    ADD CONSTRAINT `FK_scan_type_parameter_group_target_5`
+        FOREIGN KEY (`CenterID`) REFERENCES `psc` (`CenterID`);
+
+
+-- Create a new table that will be used instead of mri_protocol to identify a scan type
+
+CREATE TABLE `scan_type_parameter` (
+   `ScanTypeParameterID`      INT(11) UNSIGNED NOT NULL auto_increment,
+   `ScanTypeParameterGroupID` INT(4)  UNSIGNED NOT NULL,
+   `ScanType`   INT(10) UNSIGNED NOT NULL,
+   `HeaderName` VARCHAR(255)  NOT NULL,
+   `ValidMin`   DECIMAL(10,4) DEFAULT NULL,
+   `ValidMax`   DECIMAL(10,4) DEFAULT NULL,
+   `ValidRegex` VARCHAR(244)  DEFAULT NULL,
+   PRIMARY KEY  (ScanTypeParameterID),
+   KEY `FK_scan_type_parameter_group_ID_1` (`ScanTypeParameterGroupID`),
+   CONSTRAINT FK_scan_type_parameter_group_ID_1
+       FOREIGN KEY (ScanTypeParameterGroupID)
+           REFERENCES scan_type_parameter_group (ScanTypeParameterGroupID)
+) ENGINE=InnoDB AUTO_INCREMENT=1000 DEFAULT CHARSET=utf8;
+
+
+-- Create a new table that will be used instead of mri_protocol to identify a scan type
+CREATE TABLE scan_type_identification_failure (
+    `ScanTypeIdentificationFailureID` INT(10) UNSIGNED NOT NULL AUTO_INCREMENT,
+    `ScanTypeParameterGroupID`        INT(4)  UNSIGNED DEFAULT NULL,
+    `TarchiveID`      INT(11)        DEFAULT NULL,
+    `CandID`          INT(6)         DEFAULT NULL,
+    `PSCID`           VARCHAR(255)   DEFAULT NULL,
+    `SeriesUID`       VARCHAR(64)    DEFAULT NULL,
+    `TimeRun`         DATETIME       DEFAULT NULL,
+    `ScanLocation`    VARCHAR(255)   DEFAULT NULL,
```

Comment:
Maybe this should be more specific? Is it the DICOM or the MINC?

Reply:
I removed the MINC from the field cause it could be a MINC file, a NIfTI file, a NRRD file or any other image file type. Not the DICOMs though. I could modify to `ImageFileLocation` maybe?

Reply:
as discussed during our meeting. Given that everything DICOM is named tarchive* in the DB, keeping the name generic is fine since scanLocation can refer to many file types. Will keep `ScanLocation` for now.